idea/testData/resolve/references/JavaParameter.kt

```diff
@@ -1,6 +1,9 @@
+// IGNORE_FIR
+// TODO: This call with named argument to Java function causes resolve to fail. As a result the calls' argument list is not
+//  `FirResolvedArgumentList`, which causes resolve to fail. We may need something like a `FirPartialResolvedArgumentList` that still
```

Comment:
I'm okay with disabling a particular test with a note.

But, I'm not sure the approach you mentioned here. If it didn't become `FirResolvedArgumentList`, isn't it still `FirArgumentList` with argument mapping? Below, there is nothing more than that wrong named argument. Then, in `FirPartialResolvedArgumentList` that holds (partial) resolved mapping would be simply empty I think, and then... I don't get how the remaining resolution can go further with that and make the call "resolved" again.

If it won't be resolved with partially resolved argument list (because there would be nothing partially resolved), I wonder there could be something we can do right now instead of adding new concept. E.g., there would be some place in FIR IDE that gives up visiting argument lists if, e.g., the call is not resolved, or argument list is not resolved, etc.?

Reply:
The `FirArgumentListImpl` does not contain a mapping from args to params. We can augment it with such a mapping to supercede the "partial" concept.

The IDE logic that does the resolve is in `org.jetbrains.kotlin.idea.references.FirReferenceResolveHelper#resolveSimpleNameReference$idea_frontend_fir`, which eventually calls `org.jetbrains.kotlin.fir.expressions.FirExpressionUtilKt#getArgumentMapping` to get the mapping

Reply:
https://github.com/JetBrains/kotlin/pull/4188 adds `FirPartiallyResolvedArgumentList` to solve this. @darthorimar let me know if you prefer to have the content of https://github.com/JetBrains/kotlin/pull/4188 inside this PR.